### PR TITLE
Add mysql literals (keywords + functions) completion

### PR DESCRIPTION
**Pull request by KevinMicrosoft**
* switched metadata_executor back to cursor()
* created a mysqlcompleter, mysqlliterals, mysqlcompletion, and new parent completer mycompleter
* adjusted completion_refresher to select correct completer
* added pg and mysql lightweight metadata executors
* fixed mysql schemata and search_path handling to accommodate mysql classes

### CODE REVIEW

File: ossdbtoolsservice/language/metadata_executor.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def _columns(self, kinds=('r', 'v', 'm')):
         :return: list of (schema_name, relation_name, column_name, column_type) tuples
         """
 
-        if self.conn.server_version >= 80400:
+        if self.conn.connection.server_version >= 80400:
```
Comment (swjain23) [inline ossdbtoolsservice/language/metadata_executor.py:93]: Is this query PG specific? What about the versions?

File: ossdbtoolsservice/language/completion/mysql_completion.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from prompt_toolkit.completion import Completion    # noqa
+
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysql_completion.py:7]: remove extra line

File: ossdbtoolsservice/language/completion/mysql_completion.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+class MySQLCompletion(Completion):
+    """
+    This class extends prompt_toolkit.completion class to include specific fields (eg. schema)
+    while sending completions based on token text in pgcompleter.
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysql_completion.py:12]: token text in mysqlcompleter

File: ossdbtoolsservice/language/operations_queue.py [hunk 4]
```diff
@@ -8,11 +8,11 @@
 import threading
 from logging import Logger  
 from queue import Queue
+from prompt_toolkit.completion import Completer
```
Comment (swjain23) [inline ossdbtoolsservice/language/operations_queue.py:11]: Sort them alphabetically

File: ossdbtoolsservice/language/completion/mysqlcompleter.py [hunk 5]
```diff
@@ -0,0 +1,991 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import List
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysqlcompleter.py:6]: sort them alphabetically and put import statements first and then from statements

File: ossdbtoolsservice/language/completion/mysqlcompleter.py [hunk 6]
```diff
@@ -0,0 +1,991 @@
...
+
+    def get_schema_matches(self, suggestion, word_before_cursor):
+        schema_names = self.dbmetadata['tables'].keys()
+
+        # Unless we're sure the user really wants them, hide schema names
+        # starting with pg_, which are mostly temporary schemas
+        if not word_before_cursor.startswith('pg_'):
+            schema_names = [s
+                            for s in schema_names
+                            if not s.startswith('pg_')]
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysqlcompleter.py:692]: This is pg specific logic

File: ossdbtoolsservice/language/completion/mysqlcompleter.py [hunk 7]
```diff
@@ -0,0 +1,991 @@
...
+        prio2 = 0 if tbl.schema else 1
+        return Candidate(item, synonyms=synonyms, prio2=prio2, display=display, schema=tbl.schema_name)
+
+    def get_table_matches(self, suggestion, word_before_cursor, alias=False):
+        tables = self.populate_schema_objects(suggestion.schema, 'tables')
+        tables.extend(SchemaObject(tbl.name) for tbl in suggestion.local_tables)
+
+        # Unless we're sure the user really wants them, don't suggest the
+        # pg_catalog tables that are implicitly on the search path
+        if not suggestion.schema and (
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysqlcompleter.py:790]: This is also pg specific code, we should not add it if we are not using it as part of our literal completion. This is a huge file, we can miss stuff that is not needed if we just copy the whole file

File: ossdbtoolsservice/language/completion/mysql_completion.py (hunk 2 above)
Comment (KevinMicrosoft) [reply ossdbtoolsservice/language/completion/mysql_completion.py:7]: Which line? The standard in our codebase seems to be one line empty lines before imports and two before the class.

File: ossdbtoolsservice/language/operations_queue.py (hunk 4 above)
Comment (KevinMicrosoft) [reply ossdbtoolsservice/language/operations_queue.py:11]: Should I sort by prompt_toolkit or Completer?

File: ossdbtoolsservice/language/metadata_executor.py (hunk 1 above)
Comment (KevinMicrosoft) [reply ossdbtoolsservice/language/metadata_executor.py:93]: It is, I have moved this code to a separate PG  file.

File: ossdbtoolsservice/language/completion/mysqlcompleter.py [hunk 8]
```diff
@@ -0,0 +1,950 @@
...
+        alias = self.generate_aliases
+        s = suggestion
+        t_sug = Table(s.schema, s.table_refs, s.local_tables)
+        v_sug = View(s.schema, s.table_refs)
+        f_sug = Function(s.schema, s.table_refs, usage='from')
+        return (
+            self.get_table_matches(t_sug, word_before_cursor, alias)
+            + self.get_view_matches(v_sug, word_before_cursor, alias)
+            + self.get_function_matches(f_sug, word_before_cursor, alias)
+        )
```
Comment (swjain23) [inline ossdbtoolsservice/language/completion/mysqlcompleter.py:700]: Are we using any of these get_from_clause_item_matches method? or other get_ methods above? I don't know what they are doing so should not be part of this PR. We shouldn't add anything extra that we do not understand or need for mysql literals.

Comment (swjain23) [reply ossdbtoolsservice/language/completion/mysqlcompleter.py:700]: I think you can get rid of 80% of this file. We will only add stuff when needed.

File: ossdbtoolsservice/language/query/mysql_lightweight_metadata.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from logging import Logger  # noqa
+
+from ossdbtoolsservice.driver import ServerConnection
+from ossdbtoolsservice.language.completion.packages.parseutils.meta import ColumnMetadata, ForeignKey, FunctionMetadata     # noqa
+
+class MySQLLightweightMetadata:
+
```
Comment (swjain23) [inline ossdbtoolsservice/language/query/mysql_lightweight_metadata.py:12]: Not sure what this comment means?

File: ossdbtoolsservice/language/query/mysql_lightweight_metadata.py [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    # The boolean argument to the current_schemas function indicates whether
+    # implicit schemas, e.g. pg_catalog
+    search_path_query = '''
```
Comment (swjain23) [inline ossdbtoolsservice/language/query/mysql_lightweight_metadata.py:15]: Do we need search_path_query in MySQL?
